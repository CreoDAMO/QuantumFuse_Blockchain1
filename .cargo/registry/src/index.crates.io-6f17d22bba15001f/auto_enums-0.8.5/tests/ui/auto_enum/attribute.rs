// SPDX-License-Identifier: Apache-2.0 OR MIT

use auto_enums::auto_enum;

#[auto_enum(Iterator)]
fn unexpected_token_in_never(x: usize) -> impl Iterator<Item = i32> {
    match x {
        0 => 1..8,
        #[never(foo)] //~ ERROR unexpected token
        1 => panic!(),
        _ => (0..2).map(|x| x + 1),
    }
}

#[auto_enum(Iterator)]
fn unexpected_token_in_nested(x: usize) -> impl Iterator<Item = i32> {
    match x {
        0 => 1..8,
        #[nested(foo)] //~ ERROR unexpected token
        1 => panic!(),
        _ => (0..2).map(|x| x + 1),
    }
}

#[auto_enum(Iterator)]
fn removed_rec(x: usize) -> impl Iterator<Item = i32> {
    match x {
        0 => 1..8,
        #[rec] //~ ERROR #[rec] has been removed and replaced with #[nested]
        1 => panic!(),
        _ => (0..2).map(|x| x + 1),
    }
}

fn main() {}
