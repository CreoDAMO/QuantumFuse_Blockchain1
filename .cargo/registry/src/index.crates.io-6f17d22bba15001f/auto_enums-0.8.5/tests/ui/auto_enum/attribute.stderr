error: unexpected token in attribute
 --> tests/ui/auto_enum/attribute.rs:9:16
  |
9 |         #[never(foo)] //~ ERROR unexpected token
  |                ^

error: unexpected token in attribute
  --> tests/ui/auto_enum/attribute.rs:19:17
   |
19 |         #[nested(foo)] //~ ERROR unexpected token
   |                 ^

error: #[rec] has been removed and replaced with #[nested]
  --> tests/ui/auto_enum/attribute.rs:29:9
   |
29 |         #[rec] //~ ERROR #[rec] has been removed and replaced with #[nested]
   |         ^^^^^^
